Sources/Web3Core/KeystoreManager/BIP44.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  BIP44.swift
+//  Created by Alberto Penas Amor on 15/12/22.
+//
+
+import Foundation
+
+public protocol BIP44 {
+    /**
+    Derive an `HDNode` based on the provided `path`. The function will throws `BIP44Error.warning` if it was invoked with `warns` as true and the root key doesn't have a previous child with at least one transaction, using false the child node will be derived directly not throwing. This function needs to query the blockchain history when `warns`is true,  so it can throw network errors.
+    - Parameter path: valid BIP32 path.
+    - Parameter warns: true to be warned about following BIP44 standard, false otherwise.
+    - Throws: `BIP44Error.warning` if the child key shouldn't be used according to BIP44 standard.
+    - Returns: an HDNode child key for the provided `path` if it can be created, otherwise nil
+    */
+    func derive(path: String, warns: Bool, transactionChecker: TransactionChecker) async throws -> HDNode?
```

Comment:
I like it more than warns, but it feels like not perfect. The point is that it will throw but it is not an error, you could derive the key anyway. 
I will propose some names, if none of them are better than your proposal I will change it :)
`throwIfDisallowed`
`throwOnWarning`

Reply:
Agree that it's not an error. `throwOnWarning` sounds reasonable. Good suggestion!